relationalai-sdk-snowflake/relationalai/snowflake_sdk/api.py

```diff
@@ -0,0 +1,269 @@
+    return session.sql(f"select DELETE_RAI_ENGINE('{engine}') as status").collect()
+
+
+def get_engine(session: Session, engine: str) -> DataFrame:
+    return session.sql(f"""
+        select
+            res:id::string           as id,
+            res:account_name::string as account_name,
+            res:created_by::string   as created_by,
+            res:created_on::string   as created_on,
+            res:name::string         as name,
+            res:region::string       as region,
+            res:size::string         as size,
+            res:state::string        as state
+        from
+            (select GET_RAI_ENGINE('{engine}') as res);
+    """)
+
+
+def list_engines(session: Session) -> DataFrame:
+    return session.sql(f"""
+        select
+            value:id::string           as id,
+            value:account_name::string as account_name,
+            value:created_by::string   as created_by,
+            value:created_on::string   as created_on,
+            value:name::string         as name,
+            value:region::string       as region,
+            value:size::string         as size,
+            value:state::string        as state
+        from
+            (select LIST_RAI_ENGINES() as res),
+            lateral flatten (input => res);
+""")
+
+
+def use_engine(session: Session, engine: str) -> List[Row]:
+    res = session.sql(f"CALL USE_RAI_ENGINE('{engine}')").collect()
+
+    if res[0][0] != engine:
+        rsp = json.loads(res[0][0])
+
+        if not rsp["success"]:
+            raise Exception(rsp["message"])
+
+    return res
+
+
+def get_current_engine(session: Session) -> DataFrame:
+    return session.sql("select CURRENT_RAI_ENGINE() as current_engine")
+
+
+#################################
+# Transaction
+#################################
+
+
+def exec(session: Session, database: str, engine: str, query: str, data=None, readonly: bool = True) -> DataFrame:
+    return session.sql(f"select EXEC('{database}', '{engine}', '{query}', {data if data else 'null'}, {readonly})")
+
+
+def exec_into(session: Session, database: str, engine: str, query: str, warehouse: str, target: str, data=None, readonly: bool = True) -> DataFrame:
+    return session.sql(f"select EXEC_INTO('{database}', '{engine}', '{query}', '{data if data else 'null'}', {readonly}, '{warehouse}', '{target}')")
+
+
+#################################
+# Model
+#################################
+
+def load_model(session: Session, database: str, engine: str, name: str, path: str) -> List[Row]:
+    return session.sql(f"select LOAD_MODEL('{database}', '{engine}', '{name}', '{path}')").collect()
+
+
+def load_model_code(session: Session, database: str, engine: str, name: str, code: str) -> List[Row]:
+    return session.sql(f"select LOAD_MODEL_CODE('{database}', '{engine}', '{name}', '{code}')").collect()
+
+
+def load_model_query(session: Session, name: str, path: str) -> List[Row]:
```

Comment:
this is just used by load_model, I don't think we need to expose it

Reply:
Removed